doc/nrf/config_and_build/configuring_app/advanced_building.rst

```diff
@@ -0,0 +1,109 @@
+.. _building_advanced:
+
+Advanced building procedures
+############################
+
+.. contents::
+   :local:
+   :depth: 2
+
+You can customize the basic :ref:`building <building>` procedures in a variety of ways, depending on the configuration of your project.
+
+.. _gs_modifying_build_types:
+.. _modifying_build_types:
+
+Configuring build types
+***********************
+
+.. build_types_overview_start
+
+When the ``CONF_FILE`` variable contains a single file and this file follows the naming pattern :file:`prj_<buildtype>.conf`, then the build type will be inferred to be *<buildtype>*.
+The build type cannot be set explicitly.
+The *<buildtype>* can be any string, but it is common to use ``release`` and ``debug``.
+
+For information about how to set variables, see :ref:`zephyr:important-build-vars` in the Zephyr documentation.
+
+The following software components can be made dependent on the build type:
+
+* The Partition Manager's :ref:`static configuration <ug_pm_static>`.
+  When the build type has been inferred, the file :file:`pm_static_<buildtype>.yml` will have precedence over :file:`pm_static.yml`.
+* The :ref:`child image Kconfig configuration <ug_multi_image_permanent_changes>`.
+  The child image Kconfig overlay file is named :file:`child_image/<child_image_name>.conf` instead of :file:`prj.conf`, but otherwise follows the same pattern as the parent Kconfig.
```

Comment:
Don't quite understand why child image Kconfig overlay is compared to `prj.conf` as these are two different things.
How about something like:
When the build type has been inferred, the child image Kconfig overlay file is searched at :file:`child_image/<child_image_name>_<buildtype>.conf`

Btw, given how complex child image feature is maybe it's better not repeat partial information from the multi image page and just state that "certain child image configuration files located at :file:`child_image/` directory can be defined per build type.

Reply:
Both build types and child images are going to be reworked soon by @nordicjm , so this is not a critical edit, but I opted for option 2 and also moved some info to multi-image docs.